Description: fix cython module so extension for python3.3
 imp.get_suffixes() returns all kind of junk, the first is not correct anymore
Origin: backported from a1daa0286a867cda86e728e177d72a645b72c634
Author: Julian Taylor <jtaylor.debian@googlemail.com>

diff --git a/IPython/extensions/cythonmagic.py b/IPython/extensions/cythonmagic.py
index 94f5e9c..0f539d2 100644
--- a/IPython/extensions/cythonmagic.py
+++ b/IPython/extensions/cythonmagic.py
@@ -146,7 +146,18 @@ class CythonMagics(Magics):
         ctx = Context(cython_include_dirs, default_options)
         key = code, sys.version_info, sys.executable, Cython.__version__
         module_name = "_cython_magic_" + hashlib.md5(str(key).encode('utf-8')).hexdigest()
-        so_ext = [ ext for ext,_,mod_type in imp.get_suffixes() if mod_type == imp.C_EXTENSION ][0]
+
+        dist = Distribution()
+        config_files = dist.find_config_files()
+        try:
+            config_files.remove('setup.cfg')
+        except ValueError:
+            pass
+        dist.parse_config_files(config_files)
+        build_extension = build_ext(dist)
+        build_extension.finalize_options()
+
+        so_ext = build_extension.get_ext_filename('')
         module_path = os.path.join(lib_dir, module_name+so_ext)
 
         if not os.path.exists(lib_dir):
@@ -168,15 +179,6 @@ class CythonMagics(Magics):
                 extra_compile_args = args.compile_args,
                 libraries = args.lib,
             )
-            dist = Distribution()
-            config_files = dist.find_config_files()
-            try: 
-                config_files.remove('setup.cfg')
-            except ValueError:
-                pass
-            dist.parse_config_files(config_files)
-            build_extension = build_ext(dist)
-            build_extension.finalize_options()
             try:
                 build_extension.extensions = cythonize([extension], ctx=ctx, quiet=quiet)
             except CompileError:
